app/services/testing_ground/tree_sampler.rb

```diff
@@ -0,0 +1,31 @@
+class TestingGround
+  class TreeSampler
+    def initialize(networks)
+      @networks = networks
+    end
+
+    def sample(resolution)
+      Hash[@networks.each_pair.map do |carrier, graph|
+        graph.nodes.each do |node|
+          node.set(:load, downsample(node.load, resolution))
+        end
+
+        [carrier, GraphToTree.convert(graph)]
+      end]
+    end
+
+    private
+
+    def downsample(node_load, resolution)
+      if resolution == :low
+        node_load.each_slice(96).map do |loads|
+          absolutes = loads.map(&:abs)
+
+          loads[absolutes.index(absolutes.max)]
```

Comment:
This could be simplified as `loads.max_by(&:abs)`. [Enumerable#max_by](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-max_by).